spec/controllers/target_controller_spec.rb

```diff
@@ -23,10 +25,59 @@
   end
 
   describe 'GET index' do
-    it 'renders a list of targets' do
-      targets = create_list(:target, 2)
-      get :index
-      expect(assigns(:targets)).to match_array(targets)
+    context 'when the user is logged in' do
+      login_user
+
+      it 'renders a list of targets' do
+        targets = create_list(:target, 2)
+        get :index
+        expect(assigns(:targets)).to match_array(targets)
+      end
+
+      context 'JSON format' do
+        login_user
+        let!(:target_1) { create :target }
+        let!(:target_2) { create :target }
+
+        before :each do
+          get :index, format: :json
+          @parsed_body = JSON.parse(response.body)
+        end
+
+        it 'renders a list of targets' do
+          expect(@parsed_body.length).to eq(2)
+        end
+
+        it 'renders the edit url for each target' do
+          expect(@parsed_body[0]['edit_url']).to eq(edit_target_url(target_1))
```

Comment:
I think that we can use `expect(..).to include(..)` in cases like this one. That syntax makes it more clear what you are testing. It also allows you to 1. check multiple values without adding too much noise, and 2. cheking only for the keys (in some scenarios you just want to know if a key is there).

For example, compare these:
```ruby
expect(@parsed_body[0]['edit_url']).to eq(edit_target_url(target_1))
# vs
expect(@parsed_body[0]).to include('edit_url' => edit_target_url(target_1))
# or
expect(@parsed_body[0])
  .to include('id' => target_1.id, 'edit_url' => edit_target_url(target_1))
# ...or
expect(@parsed_body[0]).to include('edit_url', 'name', 'id')
```

Also, sometimes you can increase the readability exploiting the `subject` (it depends on the case, you can try it and see if you like it better or not):

```ruby
let(:json_response) do
  get(:index, format: :json)
  JSON.parse(response.body)
end

it 'renders two items' do
  expect(json_response.size).to eq(2) # <- Prefer `size` over `length`
end

describe 'first item' do
  subject { json_response.first }

  it { is_expected.to include('edit_url' => edit_target_url(target_1)) }
end
```